accounts-db/src/accounts.rs

```diff
@@ -753,18 +756,24 @@ impl Accounts {
                 .filter(|(i, _)| is_storable_account(message, *i))
             {
                 if message.is_writable(i) {
-                    let is_fee_payer = i == 0;
-                    let is_nonce_account = prepare_if_nonce_account(
-                        address,
-                        account,
-                        execution_status,
-                        is_fee_payer,
-                        maybe_nonce,
-                        durable_nonce,
-                        lamports_per_signature,
-                    );
-
-                    if execution_status.is_ok() || is_nonce_account || is_fee_payer {
+                    let should_store_account = match tx_status {
+                        TransactionStatus::Successful => true,
+                        TransactionStatus::FailedWithNonce { nonce } => {
+                            let is_fee_payer = i == 0;
```

Comment:
nit: since you're touching very sensible piece of code, i'd take the chance to further simplify the code by factoring out the `address == nonce.address()` into caller by defining `let is_nonce_account = ...` just below `let is_fee_payer = ...` here.

Then, make `prepare_if_nonce_account` return `()`, which in turn make the nested `if is_fee_payer { ... }` go way.

Finally, after that, `prepare_if_nonce_account()`'s job clearer: it does its jobs only if `is_nonce_account || is_fee_payer`.

Reply:
also, rename to `post_process_failed_nonce`?